### PR TITLE
Add shpec examples

### PR DESCRIPTION
**Pull request by schorsch3000**
i've took the shellspec examples and adapted them to shpec .
there is no setup and teardown in shpec.
but since it's just bash you can do that by your self.
i'm not shore if a self-build setup & teardown should be part of the example, if you think so, i'll do it.
I also took the readme and adapted it as best as i could.
There is stuff missing, mainly the opinion part and comparison.
That said, everything that's in there should be corret.

### CODE REVIEW

**Comment by dodie** [comment]
Hi @schorsch3000 ,
sorry for the late response, I got a bit swamped with work.

In my opinion everything looks great! 👍  I did not yet try shspec yet, but soon will give it a try and I'll also add some cents to the topic.

> i'm not shore if a self-build setup & teardown should be part of the example, if you think so, i'll do it.

If it's not a big deal, I think it would be really awesome to showcase how this framework can do setup and teardown!

File: example-shpec/readme.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+✔️ What's really interesting is that despite this DSL, files are simple Bash files. It's possible mix simple
+Bash statements and functions with the DSL, so even if the DSL lacks certain features, it's usually not a limitation.
+
+⚠️ This language looks really neat and extensible, but on the other hand it's something that one have to get used to
+in order to write tests efficiently.
+
+
+## Test Discovery
+
+✔️ The test runner considers all files as tests that are in the shpec folder and their name ends in "_spec.sh".
```
Comment (dodie) [inline example-shpec/readme.md]:
```suggestion
✔️ The test runner considers all files as tests that are in the shpec folder and their name ends in `_spec.sh`.
```

File: example-shpec/readme.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+By default all test cases are executed, by calling shpec with a shpec-file as the first parameter, only this case is tested.
+
+✔️ It also expects a specific project structure, test files must be in the schema of ./shpec/**/*_shpec.sh,
```
Comment (dodie) [inline example-shpec/readme.md]:
```suggestion
✔️ It also expects a specific project structure, test files must be in the schema of `./shpec/**/*_shpec.sh`, 
```

File: example-shpec/readme.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+in order to write tests efficiently.
+
+
+## Test Discovery
+
+✔️ The test runner considers all files as tests that are in the shpec folder and their name ends in "_spec.sh".
+By default all test cases are executed, by calling shpec with a shpec-file as the first parameter, only this case is tested.
+
+✔️ It also expects a specific project structure, test files must be in the schema of ./shpec/**/*_shpec.sh, 
+custom matcher should be in ./shpec/matchers/ every thing else is up to you.
```
Comment (dodie) [inline example-shpec/readme.md]:
```suggestion
custom matcher should be in `./shpec/matchers/` every thing else is up to you.
```

File: example-shpec/readme.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+## Mocking
+
+✔️ Mocking is possible with all [three common techniques](https://github.com/dodie/testing-in-bash/tree/master/mocking):
+
+- alias
+- function export
+- PATH override
+
+Also there is [stubbing](https://github.com/rylnd/shpec#stubbing) build in.
```
Comment (dodie) [inline example-shpec/readme.md:61]: Neat 👍

File: example-shpec/readme.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+## Activity
+
+⚠️ The project is not in development right now, is seems to be in a "done enough" state.
+
+
+✔️ The project had quite some releases over the past year. The author is active answering issues and
```
Comment (dodie) [inline example-shpec/readme.md:68]: I agree, there was only a single issue since the last release.

File: example-shpec/readme.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+taking care of PRs.
+
+
+## Documentation
+
+✔️ The project test's itself, the tests are not only functional but are written to be good [examples](https://github.com/rylnd/shpec/blob/master/shpec/shpec_shpec.sh).
```
Comment (dodie) [inline example-shpec/readme.md:74]:
```suggestion
✔️ The project tests itself, the tests are not only functional but are written to be good [examples](https://github.com/rylnd/shpec/blob/master/shpec/shpec_shpec.sh).
```

**Review by dodie — COMMENTED** [review]
I've just added some minor formatting suggestions. Looks good to me! 👍

**Comment by schorsch3000** [comment]
> sorry for the late response, I got a bit swamped with work.

No problem, there is none waiting for this pr :)

> >i'm not shore if a self-build setup & teardown should be part of the example, if you think so, i'll do it.

> If it's not a big deal, I think it would be really awesome to showcase how this framework can do setup and teardown!

`shpec` is not who will do the setup & teardown, in this case, this will just be a setup & teardown bash function, totally out of scope of `shpec`. For a "this is how you can do this" scenario this would be fine.
I'm a bit packed this week, but i thing i can do this on Sunday.

**Comment by dodie** [comment]
Hi @schorsch3000 ,

I merge this PR. If you still have the time to add more examples, PRs are always welcome!
Thank you for this analysis!

Cheers,
David

**Pull request merged**